pkg/xds/server/callbacks/dataplane_callbacks.go

```diff
@@ -106,25 +107,26 @@ func (d *xdsCallbacks) OnStreamRequest(streamID core_xds.StreamID, request util_
 	// wasn't processed
 	alreadyProcessed = d.dpStreams[streamID].dp != nil
 	if alreadyProcessed {
+		d.Unlock()
 		return nil
 	}
 
-	dpStream := d.dpStreams[streamID]
-	dpStream.dp = &dpKey
-	d.dpStreams[streamID] = dpStream
-
 	activeStreams := d.activeStreams[dpKey]
-	d.activeStreams[dpKey]++
-	d.Unlock()
-
 	if activeStreams == 0 {
```

Comment:
Can we switch activeStreams to be a bool now?

Reply:
I'm changing values of this map to streamIDs

Reply:
I'm changing values of this map to the single active streamID of the paticular data plane.